### PR TITLE
MPC-58 integrate training with mlflow

### PR DESCRIPTION
**Pull request by dnerini**

### CODE REVIEW

File: rainforest/ml/rf_train.py [hunk 1]
```diff
@@ -75,6 +75,13 @@ def main():
                       help="If set to 1 (default), the input parquet files (homogeneized tables) for the ml routines will be recomputed from the current database rows"+
                       "This takes a bit of time but is needed if you updated the database and want to use the new data in the training",
                       metavar="MODELS")
+    
+    parser.add_option("-l", "--logmlflow",
```
Comment (dnerini) [inline rainforest/ml/rf_train.py:79]: we need a flag to decide whether to log the model artifact, false by default?

File: rainforest/ml/rfdefinitions.py [hunk 2]
```diff
@@ -13,11 +13,17 @@
...
 import pickle
 import gzip
+import tempfile
 import numpy as np
 from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import r2_score, mean_absolute_error, root_mean_squared_error
 import os
 from scipy.interpolate import UnivariateSpline
 from pathlib import Path
+import mlflow
```
Comment (dnerini) [inline rainforest/ml/rfdefinitions.py:23]: should we make mlflow an optional dependency?

for example
```
try:
    import mlflow

    MLFLOW_INSTALLED = True
except ImportError:
   MLFLOW_INSTALLED = False
```

**Comment by wolfidan** [comment]
I just added the possibility to log also test errrors to mlflow. Sorry for the stupid commit name.

It works by using the argument -C <number_of_crossval_iterations> in rf_train.py. Default is 0 : no test error, no cross-val

File: rainforest/ml/rfdefinitions.py (hunk 2 above)
Comment (wolfidan) [reply rainforest/ml/rfdefinitions.py:23]: good idea, we will change that

File: rainforest/ml/rf_train.py (hunk 1 above)
Comment (MicheleCattaneo) [reply rainforest/ml/rf_train.py:79]:
```
parser.add_option("-l", "--logmlflow", 
                  type="choice", 
                  choices=["none", "metrics", "all"],
                  dest="logmlflow", 
                  default="none",
                  help="Specify the logging mode for MLFlow. Choices are:" +
                       " 'none' (default, no logging), 'metrics' (log metrics only)," +
                       " or 'all' (log metrics and model)." +
                       " To log to a remote ML server, the environment variable MLFLOW_TRACKING_URI needs to be set.")
```
How about a choice like this?

**Pull request merged**